medcat/config_meta_cat.py

```diff
@@ -48,27 +58,66 @@ class Config:
 class Model(MixingConfig, BaseModel):
     """The model part of the metaCAT config"""
     model_name: str = 'lstm'
-    """NOTE: When changing model, make sure to change the tokenizer as well"""
+    """
+    Model to be used for training or predicting.
+
+    Choose from:
+        - 'bert'
+        - 'lstm'
+
+    Note:
+        When changing the model, make sure to change the tokenizer accordingly.
+    """
     model_variant: str = 'bert-base-uncased'
+    """
+    Applicable only when using BERT:
+    
+    Specifies the model variant to be used.
+    """
     model_freeze_layers: bool = True
+    """
+    Applicable only when using BERT:
+    
+    Determines the training approach for BERT.
+
+    - If True: BERT layers are frozen and only the fully connected (FC) layer(s) on top are trained.
+    - If False: Parameter-efficient fine-tuning will be applied using Low-Rank Adaptation (LoRA).
+    """
     num_layers: int = 2
+    """Number of layers in the model (both LSTM and BERT)"""
     input_size: int = 300
+    """
+    Specifies the size of the embedding layer.
+
+    Applicable only for LSTM model and ignored for BERT as BERT's embedding size is predefined.
+    """
     hidden_size: int = 300
+    """Number of neurons in the hidden layer"""
     dropout: float = 0.5
     phase_number: int = 0
-    """Indicates whether or not two phase learning is being performed.
+    """Indicates whether two phase learning is to be used for training.
+    
     1: Phase 1 - Train model on undersampled data
+    
     2: Phase 2 - Continue training on full data
-    0: None - 2 phase learning is not performed"""
+    
+    0: None - 2 phase learning is not performed
+    
+    Paper reference - https://ieeexplore.ieee.org/document/7533053"""
     category_undersample: str = ''
+    """When using 2 phase learning, this category is used to undersample the data"""
     model_architecture_config: Dict = {'fc2': True, 'fc3': False,'lr_scheduler': True}
```

Comment:
I would like this to have some documentation as well. What are the keys that are expected in this dict? What do they describe? What values can they have? Are there some that don't exist in the default value yet can be optionally added?